#153741756 Users should receive a JWT on successful signup

kensanni opened the pull request:
#### What does this PR do?
>- This PR ensures a user receives a JSON web token on successful sign-up
#### Description of Task to be completed?
>- Users once authenticated via the signup API should be returned a JWT to be required on subsequent calls to all other endpoints that require authentication.

#### How should this be manually tested?
>- Open postman, input the URL address which is "localhost:3000/api/users"
>- input the following parameters:
    name: 'string'
    username: 'string'
    email: 'string'
    password: 'string'
    avatar: 'string'

>- send the request to the server
>- you should get a response containing the JSON web token

#### Any background context you want to provide?
>- N/A

#### What are the relevant pivotal tracker stories?
>- https://www.pivotaltracker.com/n/projects/2137297/stories/153741756

#### Screenshots (if appropriate):
<img width="1125" alt="screen shot 2018-01-04 at 11 19 12 pm" src="https://user-images.githubusercontent.com/22421283/34586818-dbd0e01a-f1a5-11e7-8213-cfb8a68e7300.png">


#### Questions:
>- N/A

kensanni committed:
feature(authentication): implement jwt for sign up

Install JWT for authentication
Install express-validation for validating user data
Install chai assertion library for testing
Install chai-http assertion library for testing http apis
Update mongoose to use promises
write test for user controllor
update signup page to show validation errors
update directives file to if a user is authenticated
validate user controller

[Finishes #153741756]

Chieze-Franklin commented on .eslintrc.json:
```diff
@@ -6,6 +6,12 @@
+    "describe": false,
+    "require": false,
+    "it": false
+  },
```
This doesn't seem to be the right place to effect this change.

kensanni committed:
feature(UI): make error message uniform

[#153741756]

chore(eslint): remove globals configuration from eslint

[#153741756]

chore(cleanup): remove console.log call from functions

[#153741756]

chore(test): modify test to use mock data

[#153741756]

chore(test): add more assertion to test

[#153741756]

kensanni committed:
feature(UI): make error message uniform

[#153741756]

chore(eslint): remove globals configuration from eslint

[#153741756]

chore(cleanup): remove console.log call from functions

[#153741756]

chore(test): modify test to use mock data

[#153741756]

chore(test): add more assertion to test

[#153741756]

kensanni committed:
chore(eslint): fix eslint error

[#153741756]

kensanni committed:
chore(eslint): fix eslint error

[#153741756]

kensanni committed:
Merge branch 'feature/153741756/authenticate-user-signup-with-jwt' of https://github.com/andela/blade-runner-cfh into feature/153741756/authenticate-user-signup-with-jwt

kensanni committed:
chore(feature): edit files to randomdly assign avatars to user on signup

[#153741756]
